### PR TITLE
NCDC/NOAA Paleo harvester

### PR DESCRIPTION
**Pull request by Jkovarik**
@michaeljb Data is currently being harvested in integration, Lynn provided a few metadata updates.

### CODE REVIEW

**Comment by michaeljb** [comment]
TODO:
- [ ] use `map` instead of `each/push` in `lib/search_solr_tools/harvesters/ncdc_paleo.rb`

**Comment by Jkovarik** [comment]
@michaeljb well that's still no excuse not to use map!

File: lib/search_solr_tools/helpers/ncdc_paleo_format.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        end
+      end
+
+      def self.format_ad_time(node_text)
+        node_text =~ /START YEAR:([^*]*)AD\s*\* END YEAR:([^*]*)AD/
+        {
+          start: DateTime.strptime(Regexp.last_match(1).strip, '%Y'),
+          end: DateTime.strptime(Regexp.last_match(2).strip, '%Y')
+        }
+      end
```
Comment (michaeljb) [inline lib/search_solr_tools/helpers/ncdc_paleo_format.rb]: I find that using named groups is more clear than `Regexp.last_match`, e.g.:

```ruby
def self.format_ad_time(node_text)
  match = node_text.match(/START YEAR:(?<start>[^*]*)AD\s*\* END YEAR:(?<end>[^*]*)AD/)
  {
    start: DateTime.strptime(match[:start].strip, '%Y'),
    end: DateTime.strptime(match[:end].strip, '%Y')
  }
end
```

**Comment by michaeljb** [comment]
True :grin:

File: lib/search_solr_tools/helpers/ncdc_paleo_format.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+      def self.format_cal_yr_bp_time(node_text)
+        zero_year = 1950
+        node_text =~ /START YEAR:([^*]*)... yr BP\s*\* END YEAR:([^*]*)... yr BP/
+        {
+          start: DateTime.strptime((-(Regexp.last_match(1).strip.to_i) - zero_year).to_s, '%Y'),
+          end: DateTime.strptime((-(Regexp.last_match(2).strip.to_i) - zero_year).to_s, '%Y')
+        }
+      end
```
Comment (michaeljb) [inline lib/search_solr_tools/helpers/ncdc_paleo_format.rb]: Same named group thing as in `format_ad_time`.

File: lib/search_solr_tools/helpers/ncdc_paleo_format.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+      def self.get_temporal_duration(node)
+        range = date_range(node)
+        return nil if range.empty?
```
Comment (michaeljb) [inline lib/search_solr_tools/helpers/ncdc_paleo_format.rb]: Real nitpicky here, but `return if range.empty?` is the same as `return nil if range.empty?`. I prefer to leave `nil` out when it's implied.

**Comment by michaeljb** [comment]
OK, think my TODO list comment has everything

**Comment by michaeljb** [comment]
Oh yeah, search-interface will also need to be udpated to show NCDC on the home page, like in https://github.com/nsidc/search-interface/pull/3

**Comment by Jkovarik** [comment]
@michaeljb indeed, the interface configuration was preempted by the PO until we get this updated.     In either case shouldn't *stop* the merging of this pull request once we're ready.

File: lib/search_solr_tools/helpers/ncdc_paleo_format.rb (hunk 3 above)
Comment (Jkovarik) [inline lib/search_solr_tools/helpers/ncdc_paleo_format.rb]: +1

File: lib/search_solr_tools/helpers/ncdc_paleo_format.rb (hunk 1 above)
Comment (Jkovarik) [inline lib/search_solr_tools/helpers/ncdc_paleo_format.rb]: +1

**Comment by Jkovarik** [comment]
Updates made based on review commentary

**Comment by Jkovarik** [comment]
@michaeljb I believe this is ready to go

**Comment by michaeljb** [comment]
@Jkovarik output from Travis-CI:

```
Failures:
  1) SearchSolrTools::Harvesters::NcdcPaleo#get_docs_with_translated_entries_from_ncdc_paleo translates_a_document
     Failure/Error: expect(translated_values).to match_array values
       expected collection contained:  ["00.010101 20.150923"]
       actual collection contained:    ["00.010101 20.150924"]
       the missing elements were:      ["00.010101 20.150923"]
       the extra elements were:        ["00.010101 20.150924"]
     # ./spec/unit/harvesters/ncdc_paleo_spec.rb:57:in `block (4 levels) in <top (required)>'
     # ./spec/unit/harvesters/ncdc_paleo_spec.rb:55:in `each'
     # ./spec/unit/harvesters/ncdc_paleo_spec.rb:55:in `block (3 levels) in <top (required)>'
```

It looks like a unit test depends on the current date.

**Comment by michaeljb** [comment]
```
      'temporal' => ['00.010101 20.150923'],
      'temporal_coverages' => ['1856-01-01T00:00:00+00:00,2013-01-01T00:00:00+00:00'],
```

Based on `temporal_coverages`, I would guess the value of `temporal` should be `['18.560101 20.130101']`.

**Pull request merged**